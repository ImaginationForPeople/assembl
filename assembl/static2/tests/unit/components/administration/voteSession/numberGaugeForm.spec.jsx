import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import DumbNumberGaugeForm from '../../../../../js/app/components/administration/voteSession/numberGaugeForm';
import '../../../../helpers/setupTranslations';

describe('DumbNumberGaugeForm component', () => {
  const handleMinChange = jest.fn(() => {});
  const handleMaxChange = jest.fn(() => {});
  const handleUnitChange = jest.fn(() => {});

  it('should render a form to set up a numeral gauge', () => {
    const props = {
      minimum: 2,
      maximum: 11,
      unit: 'kms',
      handleMaxChange: handleMaxChange,
      handleMinChange: handleMinChange,
      handleUnitChange: handleUnitChange
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<DumbNumberGaugeForm {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});