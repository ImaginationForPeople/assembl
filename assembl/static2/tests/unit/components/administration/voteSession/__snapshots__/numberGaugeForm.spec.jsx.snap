// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumbNumberGaugeForm component should render a form to set up a numeral gauge 1`] = `
<div>
  <FormGroup
    bsClass="form-group"
  >
    <FormControlWithLabel
      helperUrl=""
      label="Minimum value"
      labelAlwaysVisible={false}
      onChange={[Function]}
      required={true}
      type="number"
      validationErrors={null}
      value={2}
    />
    <FormControlWithLabel
      helperUrl=""
      label="Maximum value"
      labelAlwaysVisible={false}
      onChange={[Function]}
      required={true}
      type="number"
      validationErrors={null}
      value={11}
    />
    <FormControlWithLabel
      helperUrl=""
      label="Unit"
      labelAlwaysVisible={false}
      onChange={[Function]}
      required={true}
      type="text"
      validationErrors={null}
      value="kms"
    />
  </FormGroup>
</div>
`;
